src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java

```diff
@@ -12,11 +15,32 @@ public ActorDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Actor add(Actor actor) {
-        return null;
+        if (this.factory == null) {
+            throw new DataProcessingException("Factory not initialized");
+        }
+        Session session = this.factory.openSession();
+        if (session == null) {
+            throw new DataProcessingException("Session not initialized");
+        }
+        Transaction transaction = session.beginTransaction();
+        try {
+            session.persist(actor);
+            transaction.commit();
+        } catch (Exception e) {
+            transaction.rollback();
+            throw new DataProcessingException("Error in adding actor", e);
```

Comment:
It's good to see the transaction rollback in the catch block. However, make sure that the `DataProcessingException` message is specific to the operation being performed. In this case, 'Error in adding actor' is appropriate.